File: 01 -Nivelamento/unidade00o_processoAEDIIExercicios/03 - Letras Maiúsculas em Java/Uppercase.java
public class Uppercase {
    public static boolean isUpperCase(char c) {
        if (c >= 'A' && c <= 'Z') {
            return true;
        }
        return false;
    }

    public static void main(String[] args) {
        String palavra = "";
        int count;

        while (palavra.intern() != "FIM") {
            count = 0;
            palavra = MyIO.readLine();
            
            for (int i = 0; i < palavra.length(); i = i + 1) {
                if (isUpperCase(palavra.charAt(i)) == true) {
                    count = count + 1;
                }
            }

            if (palavra.intern() != "FIM") {
                MyIO.println(count);
            }
        }
    }
}

File: 01 -Nivelamento/unidade00o_processoAEDIIExercicios/04 - Letras Maiúsculas em C-C++/Uppercase.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

bool isUpperCase(char c) {
    if (c >= 'A' && c <= 'Z') {
        return true;
    }
    return false;
}

int main() {    
    char palavra[50] = "";
    int count = 0;

    while (strcmp(palavra, "FIM") != 0) {
        count = 0;
        gets(palavra);
        for (int i = 0; i < strlen(palavra); i = i + 1) {
            if (isUpperCase(palavra[i]) == true) {
                count = count + 1;
            }
        }

        if (strcmp(palavra, "FIM") != 0) {
            printf("%d\n", count);
        }
    }
    
    return 0;
}